### PR TITLE
Route read operations on file metadata to FileMeta module

### PR DESCRIPTION
**Pull request by gshaw1997**
**Purpose of this PR**
Refactors the service to use the File Metadata module for retrieving file metadata when loading Learning Objects, downloading files, and fetching file metadata.

### CODE REVIEW

File: src/drivers/MongoDriver.ts [hunk 1]
```diff
@@ -1428,45 +1459,53 @@ export class MongoDriver implements DataStore {
...
-      throw new ResourceError('Learning Object Author not found', ResourceErrorReason.NOT_FOUND);
+        },
+      ])
+      .toArray();
+    if (results && results[0]) {
+      const object = results[0];
+      object.materials.files = object.orderedFiles;
+      delete object.orderedFiles;
+      const author = await this.fetchUser(object.authorID);
+      return this.generateLearningObject(author, object, params.full);
```
Comment (nvisal1) [inline src/drivers/MongoDriver.ts]: Why did you remove the check for the author? What if the author is undefined?

File: src/drivers/MongoDriver.ts [hunk 2]
```diff
@@ -1538,16 +1584,14 @@ export class MongoDriver implements DataStore {
...
-      const author = await this.fetchUser(object[0].authorID);
-      if (author) {
-        return this.generateReleasedLearningObject(author, object[0], params.full);
-      }
-      throw new ResourceError('Learning Object Author not found', ResourceErrorReason.NOT_FOUND);
+    if (results && results[0]) {
+      const object = results[0];
+      object.materials.files = object.orderedFiles;
+      delete object.orderedFiles;
+      const author = await this.fetchUser(object.authorID);
```
Comment (nvisal1) [inline src/drivers/MongoDriver.ts]: Why did you remove the check for the author? What if the author is undefined?

File: src/drivers/MongoDriver.ts (hunk 1 above)
Comment (gshaw1997) [reply src/drivers/MongoDriver.ts]: The author should never be undefined if the Learning Object exists, if it is, there's data inconsistency within the system, and is not a result of the requester requesting a resource that doesn't exist. I would say at that point its an Internal Service Error.

Additionally, I believe we should continue to handle throwing these sorts of errors at the interactor level, since it knows the full context of the request and can convey errors in relation to that. In this case, I think it's confusing for a client requesting a Learning Object to receive an error about a user resource not existing.

File: src/drivers/MongoDriver.ts (hunk 2 above)
Comment (gshaw1997) [reply src/drivers/MongoDriver.ts]: The author should never be undefined if the Learning Object exists, if it is, there's data inconsistency within the system, and is not a result of the requester requesting a resource that doesn't exist. I would say at that point its an Internal Service Error.

Additionally, I believe we should continue to handle throwing these sorts of errors at the interactor level, since it knows the full context of the request and can convey errors in relation to that. In this case, I think it's confusing for a client requesting a Learning Object to receive an error about a user resource not existing.
